data-prepper-api/src/main/java/org/opensearch/dataprepper/model/annotations/Required.java

```diff
@@ -0,0 +1,22 @@
+package org.opensearch.dataprepper.model.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * Annotation to represent a required property and its allowed values.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Required {
```

Comment:
Let's make the context clearer. Here are two options.

1. Make this a nested annotation in `AlsoRequires`
2. Rename to `@AlsoRequired`.

I think I like option 1.